### ISSUE TITLE
Relationship of nvidia-container-runtime and CDI

### ISSUE THREAD

**Issue by ereslibre** [NVIDIA/nvidia-container-toolkit#335]
Hello,

Thank you for all your work. I am writing to ask for a clarification in terms of the current projects and components.

Based on https://github.com/NVIDIA/nvidia-docker/issues/1268#issuecomment-632692949 and https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/arch-overview.html, it's not completely clear to me what is the future relationship between `cmd/nvidia-container-runtime{-hook,.cdi,.legacy}` within this project and CDI.

While we need to support older runtimes, it's clear that we need to provide `nvidia-container-runtime` in the meantime, but do you foresee that CDI will completely deprecate it? Or is it intended for it to cover use cases that won't be covered by CDI?

Thank you.

**Comment by ereslibre** [comment]
I also found https://forums.developer.nvidia.com/t/what-is-unique-about-the-nvidia-container-runtime-does-cdi-replace-it/265975/5 that points to the complete replacement of the nvidia-container-runtime for CDI-compliant runtimes.

**Comment by ArangoGutierrez** [comment]
You are right: Yes, CDI is the intended successor and will at some point deprecate the legacy nvidia-container-runtime wrapper.

Here's the relationship:

1. The Past (Wrapper): The nvidia-container-runtime was a "shim" created before a standard existed. It wrapped a runtime (like runc) to intercept commands and manually inject GPU support.

2. The Future (Standard): CDI (Container Device Interface) is that standard. It's a vendor-neutral CNCF specification that lets runtimes discover and inject devices themselves.

The nvidia-container-toolkit's job has evolved. Its primary role is now to use the nvidia-ctk tool to generate the CDI specification files (e.g., /var/run/cdi/nvidia.yaml).

Modern container runtimes like containerd, CRI-O, Podman, and Docker Engine (v25.0+) all have native CDI support. They read this spec file directly, which removes the need for the old wrapper.

To answer your questions directly:

* Will CDI deprecate the wrapper? Yes. 

* Does the wrapper cover use cases CDI doesn't? No. The legacy wrapper is kept only for backwards compatibility with older runtimes or to translate old user commands (like --gpus all) into modern CDI injections. But is our hope that slowly users will move to newer versions of docker/containerd/crio... and will be able to leverage CDI as the standard.

We are closing this as answered.

**Comment by ereslibre** [comment]
Thanks a lot for the super detailed and explicit answers @ArangoGutierrez, very appreciated.
